services/QuillLMS/client/app/bundles/Teacher/components/premium/school_and_district_premium_modal.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+
+import SchoolSelector, { NOT_LISTED, } from '../shared/school_selector'
+import { requestPost, requestPut, } from '../../../../modules/request';
+import { closeIcon, DropdownInput, Spinner, } from '../../../Shared/index'
+
+export const SCHOOL_SELECTION_STAGE = 'school_premium_purchase_selection_stage'
+
+const PLAN_SELECTION_STAGE_NUMBER = 1
+const SCHOOL_SELECTION_STAGE_NUMBER = 2
+
+const SchoolAndDistrictPremiumModal = ({ stripeSchoolPlan, eligibleSchools, handleAlreadyPremiumSchoolSelection, userIsSignedIn, startAtSchoolSelectionStage, customerEmail, closeModal, handleNotListedSelection, }) => {
+  const [stage, setStage] = React.useState(startAtSchoolSelectionStage ? SCHOOL_SELECTION_STAGE_NUMBER : PLAN_SELECTION_STAGE_NUMBER)
+  const [selectedSchool, setSelectedSchool] = React.useState(null)
+  const [loading, setLoading] = React.useState(startAtSchoolSelectionStage)
+
+  React.useEffect(handleStartSchoolSelectionStage, [stage])
+
+  function goToSchoolSelectionStage() {
+    setLoading(true)
+    setStage(SCHOOL_SELECTION_STAGE_NUMBER)
+  }
+
+  function handleStartSchoolSelectionStage() {
+    if (stage !== SCHOOL_SELECTION_STAGE_NUMBER) { return }
+
+    if (eligibleSchools.length === 1) {
+      const schoolIds = eligibleSchools.map(s => s.id)
+      goToStripe(schoolIds)
+    } else {
+      setLoading(false)
+    }
+  }
+
+  function goToStripe(schoolIds) {
+    const path = '/stripe_integration/subscription_checkout_sessions'
+    const data = {
+      cancel_path: 'premium',
+      customer_email: customerEmail,
+      ...(schoolIds && { school_ids: JSON.stringify(schoolIds) }),
+      stripe_price_id: stripeSchoolPlan.plan.stripe_price_id
+    }
+
+    requestPost(path, data, body => { window.location.replace(body.redirect_url) })
+  }
+
+  function goToStripeWithSelectedSchool() { goToStripe([selectedSchool.value]) }
+
+  function selectSchool(idOrType) {
+    requestPut(`${process.env.DEFAULT_URL}/select_school`, {
+      school_id_or_type: idOrType,
+    }, (body) => {
+      if (idOrType === NOT_LISTED) {
+        handleNotListedSelection()
+      } else if (body.subscription) {
+        handleAlreadyPremiumSchoolSelection()
+      } else {
+        goToStripe([idOrType])
+      }
+    });
+  }
+
+  if (loading) {
+    return (
+      <div className="modal-container school-and-district-premium-modal-container">
+        <div className="modal-background" />
+        <div className="school-and-district-premium-modal stage-two quill-modal modal-body">
+          <Spinner />
+        </div>
+      </div>
+    )
+  }
+
+  if (stage === SCHOOL_SELECTION_STAGE_NUMBER) {
+    if (eligibleSchools.length) {
```

Comment:
we could just combine these into a single if statement

Reply:
We can't, because the `else` for the second `if` should only happen if the first conditional is true (ie, if the stage is SCHOOL_SELECTION_STAGE_NUMBER but there are zero eligible schools). Your comment made me realize that these nested conditionals are confusing, so I moved out the inner conditional block into a separate function.